### PR TITLE
Add local charm inspector

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3596/
Test FAILed.

**Comment by huwshimi** [comment]
QA:
- download a charm e.g. https://api.jujucharms.com/charmstore/v4/trusty/apache2-15/archive
- drop the zip on the canvas
- the local inspector should appear
- click upload
- the charm should be uploaded
- drop the zip again
- should get an option to upgrade your service
- tick the service and click Upgrade
- the service should get upgraded

**Comment by urosj** [comment]
@huwshimi, @hatched Selection of series needs to be updated a bit. First of all, new charms will have Series field in metadata.yaml and only those series should be shown. Only in case the series attribute is not defined should we list all supported series.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3597/
Test FAILed.

File: jujugui/static/gui/src/app/components/local-inspector/local-inspector.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('local-inspector', function() {
+
+  juju.components.LocalInspector = React.createClass({
+    SERIES: [
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/local-inspector/local-inspector.js]: It sounds like this list needs to go away and for us to read the zip contents and figure out what series to deploy to from the uploaded charm.

Comment (huwshimi) [inline jujugui/static/gui/src/app/components/local-inspector/local-inspector.js]: I wonder if the getting the series should be a followup. I'll move the list of series for now and add a card for getting the value from the zip.

File: jujugui/static/gui/src/app/utils/local-charm-import-helpers.js [hunk 2]
```diff
@@ -24,32 +24,17 @@ YUI.add('local-charm-import-helpers', function(Y) {
   ns.localCharmHelpers = {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/utils/local-charm-import-helpers.js]: Doesn't bother me...

File: jujugui/static/gui/src/app/components/local-inspector/local-inspector.js (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/local-inspector/local-inspector.js]: It looks like there are quite a number of places we could use this list, I'll add a card to create a single source of series.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3610/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3611/
Test PASSed.

**Comment by hatched** [comment]
:+1: QA :ok:

**Comment by kadams54** [comment]
:+1: from me.

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**